src/ios/CDVInAppBrowser.m

```diff
@@ -937,6 +952,25 @@ - (void)webView:(UIWebView*)theWebView didFailLoadWithError:(NSError*)error
     [self.navigationDelegate webView:theWebView didFailLoadWithError:error];
 }
 
+#pragma mark NSURLConnectionDataDelegate
+
+- (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticationChallenge:(NSURLAuthenticationChallenge *)challenge {
```

Comment:
Could you provide a high level description of what is happening here? Are we jacking into an authentication request to store the base URL for usage later?